staging/src/k8s.io/apiserver/pkg/storage/value/metrics_test.go

```diff
@@ -30,63 +29,97 @@ import (
+	failedPreconditionErrTransformer := PrefixTransformer{Prefix: []byte("k8s:enc:kms:v1:"), Transformer: &testTransformer{err: failedPreconditionErr}}
+	internalErrTransformer := PrefixTransformer{Prefix: []byte("k8s:enc:kms:v1:"), Transformer: &testTransformer{err: internalErr}}
+	okTransformer := PrefixTransformer{Prefix: []byte("k8s:enc:kms:v1:"), Transformer: &testTransformer{from: []byte("value")}}
+
 	testCases := []struct {
 		desc    string
+		prefix  Transformer
 		metrics []string
-		error   error
 		want    string
 	}{
 		{
-			desc: "non-status error",
+			desc:   "non-status error",
+			prefix: NewPrefixTransformers(nil, nonStatusErrTransformer),
+			metrics: []string{
+				"apiserver_storage_transformation_operations_total",
+				"apiserver_storage_transformation_failures_total",
+			},
+			want: `
+# HELP apiserver_storage_transformation_failures_total [ALPHA] (Deprecated) Total number of failed transformation operations.
+# TYPE apiserver_storage_transformation_failures_total counter
+apiserver_storage_transformation_failures_total{transformation_type="from_storage"} 1
+apiserver_storage_transformation_failures_total{transformation_type="to_storage"} 1
+# HELP apiserver_storage_transformation_operations_total [ALPHA] Total number of transformations.
+# TYPE apiserver_storage_transformation_operations_total counter
+apiserver_storage_transformation_operations_total{status="Unknown",transformation_type="from_storage",transformer_prefix="k8s:enc:kms:v1:"} 1
+apiserver_storage_transformation_operations_total{status="Unknown",transformation_type="to_storage",transformer_prefix="k8s:enc:kms:v1:"} 1
+`,
+		},
+		{
+			desc:   "ok",
+			prefix: NewPrefixTransformers(nil, okTransformer),
 			metrics: []string{
 				"apiserver_storage_transformation_operations_total",
 				"apiserver_storage_transformation_failures_total",
 			},
-			error: errors.New("foo"),
 			want: `
-  # HELP apiserver_storage_transformation_failures_total [ALPHA] (Deprecated) Total number of failed transformation operations.
-  # TYPE apiserver_storage_transformation_failures_total counter
-  apiserver_storage_transformation_failures_total{transformation_type="encrypt"} 1
-	# HELP apiserver_storage_transformation_operations_total [ALPHA] Total number of transformations.
-  # TYPE apiserver_storage_transformation_operations_total counter
-  apiserver_storage_transformation_operations_total{status="Unknown",transformation_type="encrypt",transformer_prefix="k8s:enc:kms:v1:"} 1
+# HELP apiserver_storage_transformation_operations_total [ALPHA] Total number of transformations.
+# TYPE apiserver_storage_transformation_operations_total counter
+apiserver_storage_transformation_operations_total{status="OK",transformation_type="from_storage",transformer_prefix="k8s:enc:kms:v1:"} 1
+apiserver_storage_transformation_operations_total{status="OK",transformation_type="to_storage",transformer_prefix="k8s:enc:kms:v1:"} 1
```

Comment:
cc @logicalhan on metrics changes

Reply:
There are no actual metrics changes.. this is just tests. Was this test not testing the thing it was supposed to, previously?